packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,150 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SchemaFactory } from "../../simple-tree/index.js";
+import { hydrate } from "./utils.js";
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * A recursive tree node with a single leaf value
+ */
+export class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
+	node: schemaFactory.optionalRecursive([() => DeepTreeNode]),
+	value: schemaFactory.optional(schemaFactory.number),
+}) {}
```

Comment:
while this does work fine, I think it would make more sense, and better align with our existing src/test/scalableTestTrees.ts to structure this schema differently:

```suggestion
	class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
		node: [() => DeepTreeNode, schemaFactory.number],
	}) {}
	{
		type _check = ValidateRecursiveSchema<typeof DeepTreeNode>;
	}
```

The use of ValidateRecursiveSchema here is optional, but it helps you debug if you get the schema wrong, and is recommended by our API docs for objectRecursive, so including it is recommended.

Then when reading the tree do `while (Tree.is(currentNode, DeepTreeNode) {`

Reply:
Updated